### ISSUE TITLE
evm: graph_node fails to parse block output in beta4

### ISSUE THREAD

**Issue by djuromatic** [DeFiCh/ain#2159]
## Summary
Hi,
I am having issue integrating [graph_node](https://github.com/graphprotocol/graph-node) with json rpc changi network.
I had a few runs on a on a beta2 i was able to collect from 0 to around 18k blocks once.
Upgraded to a beta4 i am able to fetch few blocks before indexer gets stuck trying to parse output from 

`eth_getBlockByHash: 0xdeb6c8fce76fedf73ea2286d843a129378096f0b0fb34be143ac27aca73ec45d`

graph_node error
```
DEBG Trying again after eth_getBlockByHash RPC call for block hash 0xdeb6c8fce76fedf73ea2286d843a129378096f0b0fb34be143ac27aca73ec45d failed (attempt #9) with result Err(Decoder error: Error("invalid type: number, expected a (both 0x-prefixed or not) hex string with length between (0; 16]", line: 0, column: 0)), provider: changi-rpc-0, component: EthereumPollingBlockIngestor
```

Thank You!
- 

## Steps to Reproduce

[Please use step-by bullet points to help the team reproduce the bug]

- clone [graph_node](https://github.com/graphprotocol/graph-node) repo
- cd graph_node/docker
- change rpc url in docker-compose.yml
- docker-compose up -d

## Environment

[Please fill all of the following or NA if not applicable]

- Node Version: 4.0.0-beta4
- Block height on bug if applicable: 
- TX or TX type on bug if applicable:  
- OS with version:  Ubuntu 22.04.2 LTS - jammy
- Any other relevant environment info:

**Comment by prasannavl** [comment]
Thanks @djuromatic - we'll have a look at this soon. I expect it could be the result of the bugs that were fixed since.

**Comment by madeinusmate** [comment]
Hey @prasannavl !
I'm also facing issues with graph_node - deployed on Beta 10

Aug 29 11:41:10.913 WARN Unexpected RPC error, error: Rpc(Error { code: InvalidParams, message: "invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 104", data: None }), sgd: 1, subgraph_id: QmdiDovyywExjirB5wk8UpDGLa3dG9hUtRpVMfv3zKwaNt, component: BlockStream

might be due to non compliant EIP-1898 blocks - source: https://github.com/graphprotocol/graph-node/issues/2319

**Comment by djuromatic** [comment]
@madeinusmate Hello mate, Have You mayme managed to fix your issue I am getting the same thing on 4.0.2

```
Unexpected RPC error, error: Rpc(Error { code: InvalidParams, message: "invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 108", data: None }), sgd: 2, subgraph_id: QmWViFq7AniVMBTRm5hq2yv5Cxqp5Ms1noJXhGQWPLxT2F, component: BlockStream
```


```
Trying again after eth_getLogs RPC call for block range: [85340..85340] failed (attempt #10) with result Err(Rpc(Error { code: InvalidParams, message: "invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 108", data: None })), sgd: 2, subgraph_id: QmWViFq7AniVMBTRm5hq2yv5Cxqp5Ms1noJXhGQWPLxT2F, component: BlockStream
```

**Comment by sieniven** [comment]
This issue should be resolved once this PR is ready and merged into master. Would be great to get some help in ensuring the bug is fixed on your end once that happens before the next patch is released if that is possible. Many thanks.

**Comment by madeinusmate** [comment]
> @madeinusmate Hello mate, Have You maybe managed to fix your issue I am getting the same thing on 4.0.2, 4.0.3
> 
> 
> 
> ### Changi Logs
> 
> ```
> 
> 2023-11-30T16:26:09Z [ain_grpc::rpc::eth] Error parsing "input" as "GetLogsRequest": InvalidParams(invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 100)
> 
> 2023-11-30T16:26:09Z [jsonrpsee_types::params] [next_inner] Deserialization to "ain_grpc::logs::GetLogsRequest" failed. Error: Error("invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes", line: 1, column: 100), input JSON: "{\"address\":\"0xa99f58760a6bcdc177788de4a85a15245ee543e9\",\"fromBlock\":\"0x0\",\"toBlock\":\"0x0\",\"topics\":[[\"0x2170c741c41531aec20e7c107c24eecfdd15e69c9bb0a8dd37b1840b9e0b207b\",\"0xe5ce249087ce04f05a957192435400fd97868dba0e6a4b4c049abf8af80dae78\",\"0x6edcaf6241105b4c94c2efdbf3a6b12458eb3d07be3a0e81d24b13c44045fe7a\",\"0x18e1ea4139e68413d7d08aa752e71568e36b2c5bf940893314c2c5b01eaa0c42\"]]}]"
> 
> 
> 
> ```
> 
> 
> 
> 
> 
> ### Graph logs
> 
> 
> 
> ```
> 
> Unexpected RPC error, error: Rpc(Error { code: InvalidParams, message: "invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 108", data: None }), sgd: 2, subgraph_id: QmWViFq7AniVMBTRm5hq2yv5Cxqp5Ms1noJXhGQWPLxT2F, component: BlockStream
> 
> ```
> 
> 
> 
> 
> 
> ```
> 
> Trying again after eth_getLogs RPC call for block range: [85340..85340] failed (attempt #10) with result Err(Rpc(Error { code: InvalidParams, message: "invalid type: sequence, expected a (both 0x-prefixed or not) hex string or byte array containing 32 bytes at line 1 column 108", data: None })), sgd: 2, subgraph_id: QmWViFq7AniVMBTRm5hq2yv5Cxqp5Ms1noJXhGQWPLxT2F, component: BlockStream
> 
> ```
> 
> 
> 
> 
> 
> 

Not yet, will give it another shot in the next week

**Comment by sieniven** [comment]
Cheers. Many thanks for verifying!

**Comment by madeinusmate** [comment]
@djuromatic i'm still hitting the same roadblock - did you modify in anyway the graph-node settings? (beside setting the DMC RPC node)

**Comment by djuromatic** [comment]
> @djuromatic i'm still hitting the same roadblock - did you modify in anyway the graph-node settings? (beside setting the DMC RPC node)

No i have used it as it is no changes made.

What version of dmc node?

**Comment by madeinusmate** [comment]
Mmm I'm testing using the public rpc https://eth.testnet.ocean.jellyfishsdk.com/ and myDefichain one too. maybe they are not updated? 

Did you spin up your own node?
